connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorConnectorsIntegrationTest.java

```diff
@@ -150,23 +152,73 @@ public void setup() throws IOException {
         mm2Props.put("backup.bootstrap.servers", backup.kafka().bootstrapServers());
         mm2Config = new MirrorMakerConfig(mm2Props);
 
+        // we wait for the connector and tasks to come up for each connector, so that when we do the
+        // actual testing, we are certain that the tasks are up and running; this will prevent
+        // flaky tests where the connector and tasks didn't start up in time for the tests to be
+        // run
         backup.configureConnector("MirrorSourceConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorSourceConnector.class));
+        waitForConnectorToBeUp(backup, "MirrorSourceConnector");
```

Comment:
Do we need to do these one at a time or could we start all these connectors together and then wait for them to come up? Does it matter?

Reply:
We definitely can start all the connectors together and then wait for them to come up. In either case I don't think it necessarily matters. What I did notice though was that once `configureConnector` is called, if you hit the REST endpoint that lists all the connectors, it almost immediately lists that connector that was just configured. However, what the bottle neck here seems to be is the actual REST url being served for that specific connector takes a while to come up. Since we have to check if the tasks for each connector are up one at a time, I figured for consistency's sake in the code we could also check for the connector to be up one at time. What do you think is the better way to go here?